### DISCUSSION TITLE
Populate a form from a JSON endpoint

### DISCUSSION THREAD

**Discussion by AndreKR** [Budibase/budibase#11510] [Help] [answered]
I have an API endpoint that returns a simple JSON:

```
{
  "foo": 3,
  "bar": 6
}
```
I am in control of the API, so I could change the structure if necessary.

I want to show those in a form. (And on change trigger another API call, but I can't even figure out how to display the current values.)

![image](https://github.com/Budibase/budibase/assets/1188538/a42d11d5-2c0b-4f9b-89cd-c53e0fa8f706)

I added a REST Query, added a form, set its schema to the query and added a number field to the form. But the "Field" parameter of the number field has no options and if I type `foo` in there manually the number field doesn't show up. Do I need a Data Provider? A Repeater?

**Comment by AndreKR** [comment, marked as answer]
I think I figured it out.

I need a Repeater Block because only those can be referred to by bindings (`{{ ... }}`).

The "Field" parameter of the field is not, as I thought at first, the (API) field that this field is bound to. Instead it is an arbitrary identifier for the field that can be used to refer to this field from other components.

**Reply by melohagan** [reply]
Hey @AndreKR 

There is also a popular feature request for dynamic form inputs / building forms from a JSON schema: 

- https://github.com/Budibase/budibase/issues/10860
- https://github.com/Budibase/budibase/issues/5658
